pkg/rbac/namespacedAuthorization.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Contributors to the Open Cluster Management project
+package rbac
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+type userData struct {
+	err        error
+	namespaces []string
+	updatedAt  time.Time
+	lock       sync.Mutex
+}
+
+func (cache *Cache) NamespacedResources(ctx context.Context, clientToken string) (*userData, error) {
+	uid := cache.tokenReviews[clientToken].tokenReview.Status.User.UID //get uid from tokenreview
```

Comment:
1. I'd rename this function to `GetUserData()`. Remember that we will be collecting a lot more data for the user, not just the namespaces.
2. You could make this change to simplify the tests. 
```suggestion
func (cache *Cache) NamespacedResources(ctx context.Context, uid string) (*userData, error) {
```